nixos/lib/test-driver/test_driver/machine.py

```diff
@@ -740,6 +740,74 @@ def wait_for_shutdown(self) -> None:
             self.booted = False
             self.connected = False
 
+    def wait_for_qmp_event(self, event_model_filter: dict[str, Any] | None = None, timeout: int = 60 * 10) -> dict[str, Any]:
+        """
+        Wait for a QMP event that is modelled after the passed (or not) `event_model_filter`
+        For example, if you are waiting for { 'event': 'SHUTDOWN', 'data': { 'reason': 'guest-reset' } }
+
+        You can pass that exact dictionary and it will return your the actual event, which may contain more data.
+
+        It will skip all events received in the meantime, if you want to keep them,
+        you have to do the bookkeeping yourself and store them somewhere.
```

Comment:
There is a lot of semantics in this function that thins down the different possible use cases for this function, while at the same time a function that simply consumes a predicate function would have multiple advantages:

- less (event-specific) code in the machine class
- more freedom for the calling user
- generally more possibilities regarding future use cases.

Reply:
It'd feel like using the list `filter` function in Nix.
Also
- easier to understand because the important logic isn't hidden away in the test framework

Reply:
Yep, agreed.

Reply:
Done.